src/util/expr_cast.h

```diff
@@ -107,6 +107,32 @@ auto expr_try_dynamic_cast(TExpr &base)
   return ret;
 }
 
+/// \brief Try to cast a generic exprt to a specific derived class.
+/// \tparam T: The type to cast the `base` param to.
```

Comment:
Nit pick: I'd use `\p base` (it will be rendered the same way, but conveys the fact that this is a parameter)